clients/src/main/java/org/apache/kafka/common/record/DefaultRecord.java

```diff
@@ -349,7 +351,16 @@ private static DefaultRecord readFrom(ByteBuffer buffer,
 
         if (numHeaders == 0)
             return new DefaultRecord(sizeInBytes, attributes, offset, timestamp, sequence, key, value, Record.EMPTY_HEADERS);
+        try {
```

Comment:
Thanks for clarifying. Sometimes it is hard to get the exact meaning - especially that it is my first Kafka contribution. Will adjust the PR accordingly :)